stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java

```diff
@@ -0,0 +1,382 @@
+      hdfsConf.set(PropertyKey.Name.USER_FILE_WRITE_TYPE_DEFAULT, "CACHE_THROUGH");
+      FileSystem prepareFs = FileSystem.get(new URI(mParameters.mBasePath), hdfsConf);
+
+      // initialize the base, for only the non-distributed task (the cluster launching task)
+      Path path = new Path(mParameters.mBasePath);
+
+      // the base path depends on the operation
+      Path basePath;
+      if (mParameters.mOperation == Operation.CreateDir) {
+        basePath = new Path(path, "dirs");
+      } else {
+        basePath = new Path(path, "files");
+      }
+
+      if (mParameters.mOperation == Operation.CreateFile
+          || mParameters.mOperation == Operation.CreateDir) {
+        prepareFs.delete(basePath, true);
+        prepareFs.mkdirs(basePath);
+      } else {
+        // these are read operations. the directory must exist
+        if (!prepareFs.exists(basePath)) {
+          throw new IllegalStateException(String
+              .format("base path (%s) must exist for operation (%s)", basePath,
+                  mParameters.mOperation));
+        }
+      }
+      if (!prepareFs.isDirectory(basePath)) {
+        throw new IllegalStateException(String
+            .format("base path (%s) must be a directory for operation (%s)", basePath,
+                mParameters.mOperation));
+      }
+    }
+
+    // set hdfs conf for all test clients
+    Configuration hdfsConf = new Configuration();
+    // do not cache these clients
+    hdfsConf.set(
+        String.format("fs.%s.impl.disable.cache", (new URI(mParameters.mBasePath)).getScheme()),
+        "true");
+    for (Map.Entry<String, String> entry : mParameters.mConf.entrySet()) {
+      hdfsConf.set(entry.getKey(), entry.getValue());
+    }
+    mCachedFs = new FileSystem[mParameters.mClients];
+    for (int i = 0; i < mCachedFs.length; i++) {
+      mCachedFs[i] = FileSystem.get(new URI(mParameters.mBasePath), hdfsConf);
+    }
+  }
+
+  @Override
+  public MasterBenchTaskResult runLocal() throws Exception {
+    ExecutorService service =
+        ExecutorServiceFactories.fixedThreadPool("bench-thread", mParameters.mThreads).create();
+
+    RateLimiter rateLimiter = RateLimiter.create(mParameters.mTargetThroughput);
+
+    mFiledata = new byte[(int) FormatUtils.parseSpaceSize(mParameters.mCreateFileSize)];
+    Arrays.fill(mFiledata, (byte) 0x7A);
+
+    long durationMs = FormatUtils.parseTimeSize(mParameters.mDuration);
+    long warmupMs = FormatUtils.parseTimeSize(mParameters.mWarmup);
+    long startMs = mBaseParameters.mStartMs;
+    if (mBaseParameters.mStartMs == BaseParameters.UNDEFINED_START_MS) {
+      startMs = CommonUtils.getCurrentMs() + 1000;
+    }
+    long endMs = startMs + warmupMs + durationMs;
+    BenchContext context = new BenchContext(rateLimiter, startMs, endMs);
+    for (int i = 0; i < mParameters.mThreads; i++) {
+      service.submit(new BenchThread(context, mCachedFs[i % mCachedFs.length]));
```

Comment:
you could use the executors service's `InvokeAll` method instead? It clean up some of this code:

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html#invokeAll-java.util.Collection-long-java.util.concurrent.TimeUnit-

Reply:
done